src/chrome/content/rules/Omtrdc.net.xml

```diff
@@ -1,46 +1,34 @@
 <!--
 	For other Adobe coverage, see Adobe.xml.
 
-
 	Problematic domains:
-
 		- \w+-\w+.d[1-3].sc.omtrdc.net *
-
 	* Mismatched
 
-
 	Mixed content:
-
 		- Images from assets.omniture.com *
-
 	* Secured by us
 
-
-	If we are going to include ads and tracking info, let's at least do it by https.
-
+	If we are going to include ads and tracking info, let's at least do it by HTTPS.
 -->
-<ruleset name="omtrdc.net">
 
+<ruleset name="omtrdc.net">
 	<target host="*.sc.omtrdc.net" />
+		<test url="http://a.sc.omtrdc.net/optout.html" />
+		<test url="http://b.d1.sc.omtrdc.net/optout.html" />
+		<test url="http://c.d2.sc.omtrdc.net/optout.html" />
 	<target host="*.tt.omtrdc.net" />
-
-		<test url="http://totalsystemservices.d1.sc.omtrdc.net/optout.html" />
-		<test url="http://www.d1.sc.omtrdc.net/optout.html" />
-		<test url="http://nhkonline.sc.omtrdc.net/optout.html" />
-		<test url="http://adobe.tt.omtrdc.net/" />
-
-
-	<securecookie host="^(?:.*\.)?omtrdc\.net$" name=".+" />
+		<test url="http://a.tt.omtrdc.net/optout" />
+		<test url="http://b.tt.omtrdc.net/optout" />
+		<test url="http://c.tt.omtrdc.net/optout" />
 
 
-	<rule from="^http://(\w+)\.(\w+)\.d([1-3])\.sc\.omtrdc\.net/"
-		to="https://$1-$2.d$3.sc.omtrdc.net/" />
+	<securecookie host="^(.*\.)?omtrdc\.net$" name=".+" />
 
-		<test url="http://schindler.com.d3.sc.omtrdc.net/optout.html" />
-		<test url="http://www.totalsystemservices.d1.sc.omtrdc.net/optout.html" />
-		<test url="http://www.wackerchemie.d1.sc.omtrdc.net/optout.html" />
 
-	<rule from="^http:"
-		to="https:" />
+	<rule from="^http://([\w-]+)\.(sc|tt)\.omtrdc\.net/"
+		to="https://$1.$2.omtrdc.net/" />
 
+	<rule from="^http://([\w-]+)\.d([1-3])\.sc\.omtrdc\.net/"
+		to="https://$1.d$2.sc.omtrdc.net/" />
```

Comment:
Found that regex groups become potentially problematic over time. Could you take the possible targets and list them above? Or perhaps they should rely on the wildcard above? Otherwise that would be preferred with the trivial rule `	<rule from="^http:"
		to="https:" />` still in place.

Reply:
The wildcard `<target>` is necessary because all subdomains exist.

I don't see a way to do the `<rule>`s without regex groups. We want to redirect `http://a.sc.omtrdc.net` but not `http://a.a.sc.omtrdc.net`.